ui/src/main/java/org/pentaho/di/ui/spoon/tree/TreeManager.java

```diff
@@ -188,6 +190,11 @@ private <T> void render( TreeNode treeNode, T tree ) {
     TreeItem childTreeItem = createTreeItem( treeNode, tree );
     if ( treeNode.hasChildren() ) {
       for ( TreeNode childTreeNode : treeNode.getChildren() ) {
+        Repository repository = Spoon.getInstance().getRepository();
+        if( childTreeNode.getLabel().equals( "VFS Connections" ) && repository!=null
```

Comment:
Would checking  for an english label work in other internalization scenarios (i18n) ?

`childTreeNode.getLabel().equals( "VFS Connections" )`

Please check and verify .

Reply:
duplicate observation for item-2 from comment: https://github.com/pentaho/pentaho-kettle/pull/9219#discussion_r1517591573

Reply:
Changed the location of code, please re-review

Reply:
moved to ConnectionFolderProvider, looks good.